cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/MonitorHandlerAuthorizationTest.java

```diff
@@ -0,0 +1,355 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.services.http.handlers;
+
+import com.google.common.util.concurrent.AbstractService;
+import com.google.gson.Gson;
+import io.cdap.cdap.app.store.ServiceStore;
+import io.cdap.cdap.common.twill.MasterServiceManager;
+import io.cdap.cdap.gateway.handlers.MonitorHandler;
+import io.cdap.cdap.proto.Instances;
+import io.cdap.cdap.proto.RestartServiceInstancesStatus;
+import io.cdap.cdap.proto.SystemServiceLiveInfo;
+import io.cdap.cdap.proto.element.EntityType;
+import io.cdap.cdap.proto.id.InstanceId;
+import io.cdap.cdap.proto.id.SystemServiceId;
+import io.cdap.cdap.proto.security.ApplicationPermission;
+import io.cdap.cdap.proto.security.Authorizable;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+import io.cdap.cdap.proto.security.StandardPermission;
+import io.cdap.cdap.security.auth.context.AuthenticationTestContext;
+import io.cdap.cdap.security.authorization.InMemoryAccessController;
+import io.cdap.cdap.security.spi.authorization.UnauthorizedException;
+import io.cdap.http.HttpResponder;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpVersion;
+import org.apache.twill.api.logging.LogEntry;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Monitor handler authorization tests.
+ */
+public class MonitorHandlerAuthorizationTest {
+  private static final String SERVICE_NAME = "service";
+  private static final String INSTANCE = "";
+  private static final Principal MASTER_PRINCIPAL = new Principal("master", Principal.PrincipalType.USER);
+  private static final Principal UNPRIVILEGED_PRINCIPAL = new Principal("unprivileged",
+                                                                       Principal.PrincipalType.USER);
+
+  private static final Gson GSON = new Gson();
+
+  /**
+   * Mock class for the ServiceStore interface which does nothing and returns nothing.
+   */
+  static class NoOpServiceStore extends AbstractService implements ServiceStore {
```

Comment:
Consider using `Mockito` mocks. This would save tons of boilerplate code

Reply:
Done.